neo/app/actor/NeoActor.scala

```diff
@@ -1,63 +1,86 @@
+import scala.concurrent.{ExecutionContext, Future}
 import scala.concurrent.duration._
 
-class NeoActor(out: ActorRef, val berryClientProps: Props)(implicit ec: ExecutionContext) extends Actor with ActorLogging {
-
-  val berryClient = context.watch(context.actorOf(berryClientProps))
+class NeoActor(out: ActorRef, ws: WSClient, host: String, config: Config)
+              (implicit ec: ExecutionContext, implicit val materializer: Materializer) extends Actor with ActorLogging {
 
   import NeoActor._
-
+  import RequestType._
   implicit val timeout: Timeout = Timeout(20.minutes)
 
   override def receive: Receive = {
     case json: JsValue =>
       json.validate[UserRequest].map { userRequest =>
-        val allRequest = generateCBerryRequest(userRequest)
-        import RequestType._
-
-        // Example of sending the non-slicing query using ask pattern
-        val sampleJson = allRequest(Sample)
-        val fAnswer = (berryClient ? sampleJson).mapTo[JsValue].map(json => Json.obj("key" -> "sample", "value" -> json))
-        fAnswer pipeTo out
-
-        // Example of sending the group of slicing ones using actor stream pattern
-        val groupTimePlaceTags = Seq(ByTime, ByPlace, ByHashTag).map(allRequest(_))
-        val json = JsObject(Seq(
-          "batch" -> JsArray(groupTimePlaceTags),
-          "option" -> JsObject(Seq("sliceMillis" -> JsNumber(2000))
-          )))
-        berryClient.tell(BerryClient.Request(json, NeoTransformer("batch")), out)
+        val berryRequest = generateCBerryRequest(userRequest)
+        val url = "http://" + host + "/berry"
+        handleSamplingResponse(url, berryRequest)
+        handleSlicingResponse(url, berryRequest)
       }.recoverTotal {
         e => out ! JsError.toJson(e)
       }
   }
-}
 
-object NeoActor {
-  def props(out: ActorRef, berryClientProp: Props)(implicit ec: ExecutionContext) = Props(new NeoActor(out, berryClientProp))
+  private def handleSamplingResponse(url: String, berryRequest: Map[RequestType.Value, JsValue]): Unit = {
+    handleResponse(url, Sample, berryRequest(Sample))
+  }
 
-  case class NeoTransformer(key: String) extends BerryClient.IPostTransform {
-    override def transform(jsValue: JsValue): JsValue = {
-      Json.obj("key" -> key, "value" -> jsValue)
+  private def handleSlicingResponse(url: String, berryRequest: Map[RequestType.Value, JsValue]): Unit = {
+    val groupTimePlaceTags = Seq(ByTime, ByPlace, ByHashTag).map(berryRequest(_))
+    val batchJson = JsObject(Seq(
+      "batch" -> JsArray(groupTimePlaceTags),
+      "option" -> JsObject(Seq("sliceMillis" -> JsNumber(2000))
+      )))
+    handleResponse(url, Batch, batchJson)
+  }
+
+  private def handleResponse(url: String, requestType: RequestType.Value, requestBody: JsValue): Unit = {
+    val response : Future[StreamedResponse] =
+      ws.url(url).withMethod("POST").withBody(requestBody).stream()
+
+    response.map{ res =>
+      if(res.headers.status == 200){
+        try {
```

Comment:
At my first glance, it suggests us:
1. to use `Util.Try` instead of `try`
2. our `catch` function catches all errors, which is not a good way because we should let the fatal errors continue raising.
I'll follow up those points